apps/src/code-studio/initApp/loadApp.js

```diff
@@ -132,6 +133,10 @@ module.exports = function (callback) {
         renderAbusive();
         return $.Deferred().reject();
       }
+      if (dashboard.project.hideBecausePrivacyViolationOrProfane()) {
+        renderPrivacyProfanityViolating();
+        return $.Deferred().reject();
```

Comment:
Would be great to replace these with native (polyfilled) promises instead of jquery deferred at some point, tho prob out of scope for this pr